File: packages/shared-api/src/admin/contact-classification.ts
import type { AxiosInstance } from 'axios';
import type { ApiResponse, PaginatedResponse } from '@shared/types';

export interface ClassificationKeywordItem {
  id: string;
  keyword: string;
}

export interface ClassificationCategory {
  id: string;
  name: string;
  display_name: string;
  sort_order: number;
  keywords: string[] | null;
  keyword_ids: ClassificationKeywordItem[] | null;
}

export interface ClassificationLevel {
  id: string;
  name: string;
  display_name: string;
  sort_order: number;
  is_sendable: boolean;
  created_at: string;
  updated_at: string;
  categories: ClassificationCategory[] | null;
}

export function contactClassificationApi(client: AxiosInstance) {
  return {
    listLevels: () =>
      client.get<PaginatedResponse<ClassificationLevel>>('/api/v1/contact-classification/levels'),
    createLevel: (data: { name: string; display_name: string; sort_order?: number; is_sendable?: boolean }) =>
      client.post<ApiResponse<ClassificationLevel>>('/api/v1/contact-classification/levels', data),
    updateLevel: (id: string, data: Partial<{ name: string; display_name: string; sort_order: number; is_sendable: boolean }>) =>
      client.put<ApiResponse<ClassificationLevel>>(`/api/v1/contact-classification/levels/${id}`, data),
    deleteLevel: (id: string) =>
      client.delete<ApiResponse<{ deleted: boolean }>>(`/api/v1/contact-classification/levels/${id}`),
    createCategory: (data: { level_id: string; name: string; display_name: string; sort_order?: number }) =>
      client.post<ApiResponse<ClassificationCategory>>('/api/v1/contact-classification/categories', data),
    deleteCategory: (id: string) =>
      client.delete<ApiResponse<{ deleted: boolean }>>(`/api/v1/contact-classification/categories/${id}`),
    addKeywords: (data: { category_id: string; keywords: string[] }) =>
      client.post<PaginatedResponse<ClassificationKeywordItem>>('/api/v1/contact-classification/keywords', data),
    deleteKeyword: (id: string) =>
      client.delete<ApiResponse<{ deleted: boolean }>>(`/api/v1/contact-classification/keywords/${id}`),
  };
}

File: backend/app/api/admin/contact_classification.py
"""Admin API：联系人职位分类管理（等级 / 类别 / 关键词 CRUD）。"""

from fastapi import APIRouter, Depends

from app.core.responses import paginated_response, success_response
from app.security.dependencies import PlatformAuthContext, get_current_platform_user
from app.services.contact_classification_service import ContactClassificationService

router = APIRouter(prefix="/contact-classification", tags=["admin-contact-classification"])
service = ContactClassificationService()


# ── 等级 ──────────────────────────────────────────────────────────────────────

@router.get("/levels")
async def list_levels(context: PlatformAuthContext = Depends(get_current_platform_user)) -> dict:
    """返回完整层级树（等级 → 类别 → 关键词）。"""
    items = await service.list_levels(context.connection)
    return paginated_response(items, total=len(items))


@router.post("/levels")
async def create_level(
    payload: dict,
    context: PlatformAuthContext = Depends(get_current_platform_user),
) -> dict:
    """新建等级（name, display_name, sort_order, is_sendable）。"""
    item = await service.create_level(context.connection, payload)
    return success_response(item)


@router.put("/levels/{level_id}")
async def update_level(
    level_id: str,
    payload: dict,
    context: PlatformAuthContext = Depends(get_current_platform_user),
) -> dict:
    """更新等级属性（含 is_sendable）。"""
    item = await service.update_level(context.connection, level_id, payload)
    return success_response(item)


@router.delete("/levels/{level_id}")
async def delete_level(
    level_id: str,
    context: PlatformAuthContext = Depends(get_current_platform_user),
) -> dict:
    """删除等级（CASCADE 删除关联类别和关键词）。"""
    await service.delete_level(context.connection, level_id)
    return success_response({"deleted": True})


# ── 类别 ──────────────────────────────────────────────────────────────────────

@router.post("/categories")
async def create_category(
    payload: dict,
    context: PlatformAuthContext = Depends(get_current_platform_user),
) -> dict:
    """新建类别（level_id, name, display_name, sort_order）。"""
    item = await service.create_category(context.connection, payload)
    return success_response(item)


@router.delete("/categories/{category_id}")
async def delete_category(
    category_id: str,
    context: PlatformAuthContext = Depends(get_current_platform_user),
) -> dict:
    """删除类别（CASCADE 删除关联关键词）。"""
    await service.delete_category(context.connection, category_id)
    return success_response({"deleted": True})


# ── 关键词 ────────────────────────────────────────────────────────────────────

@router.post("/keywords")
async def add_keywords(
    payload: dict,
    context: PlatformAuthContext = Depends(get_current_platform_user),
) -> dict:
    """批量添加关键词（body: {category_id: str, keywords: string[]}）。"""
    category_id: str = payload["category_id"]
    keywords: list[str] = payload["keywords"]
    inserted = await service.add_keywords(context.connection, category_id, keywords)
    return paginated_response(inserted, total=len(inserted))


@router.delete("/keywords/{keyword_id}")
async def delete_keyword(
    keyword_id: str,
    context: PlatformAuthContext = Depends(get_current_platform_user),
) -> dict:
    """删除单个关键词。"""
    await service.delete_keyword(context.connection, keyword_id)
    return success_response({"deleted": True})
